analysis_driver/pipelines/rapid.py

```diff
@@ -0,0 +1,272 @@
+import os
+import csv
+from datetime import date
+from collections import defaultdict
+from egcg_core.config import cfg
+from egcg_core import executor, util, rest_communication
+from analysis_driver import segmentation, transfer_data
+from analysis_driver.util import bash_commands
+from analysis_driver.reader.demultiplexing_parsers import parse_interop_summary
+
+
+class RapidStage(segmentation.Stage):
+    def rapid_output_dir(self, lane):
+        return os.path.join(self.job_dir, 'rapid_analysis_%s' % lane)
+
+
+class Dragen(RapidStage):
+    def _run(self): 
+        cmds = []
+        for lane, sample in self.dataset.rapid_samples_by_lane.items():
+            # clean up any previous run on the Dragen server's scratch space
+            cmd = 'if [ -d {tmp_dir} ]; then rm -r {tmp_dir}; fi; mkdir -p {tmp_dir}; ' \
+                  'echo "Starting at $(date +%Y-%m-%d\ %H:%M:%S)"; '
+
+            # run Dragen on the lane with bam/vcf outputs, dbSNP and duplicate marking enabled
+            cmd += '{dragen} -r {ref} --bcl-input-dir {run_dir} --bcl-only-lane {lane} --output-directory {tmp_dir} ' \
+                   '--output-file-prefix {user_sample_id} --enable-variant-caller true ' \
+                   '--vc-sample-name {user_sample_id} --bcl-sample-sheet {sample_sheet} ' \
+                   '--enable-map-align-output true --enable-duplicate-marking true --dbsnp {dbsnp}; ' \
+
+            # somehow, writing the output data to scratch and then rsyncing to the main file system is faster than
+            # outputting directly
+            cmd += 'echo "Outputting data at $(date +%Y-%m-%d\ %H:%M:%S)"; ' \
+                   'rsync -rLD {tmp_dir}/ {out_dir}; rm -r {tmp_dir}; ' \
+                   'echo "Done at $(date +%Y-%m-%d\ %H:%M:%S)"'
+
+            cmds.append(
```

Comment:
We want Dragen's exit status to be returned, but the cleanup afterwards always needs to happen otherwise the server will run out of scratch space. Adding `&&` for setup commands, but keeping `;` after Dragen and capturing/returning `$?`